src/app/settings/user-billing-history.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, OnInit } from "@angular/core";
+import { Router } from "@angular/router";
+
+import { ApiService } from "jslib-common/abstractions/api.service";
+import { I18nService } from "jslib-common/abstractions/i18n.service";
+import { PlatformUtilsService } from "jslib-common/abstractions/platformUtils.service";
+import { PaymentMethodType } from "jslib-common/enums/paymentMethodType";
+import { TransactionType } from "jslib-common/enums/transactionType";
+import { BillingHistoryResponse } from "jslib-common/models/response/billingHistoryResponse";
+
+@Component({
+  selector: "app-user-billing",
+  templateUrl: "user-billing-history.component.html",
+})
+export class UserBillingHistoryComponent implements OnInit {
+  loading = false;
+  firstLoaded = false;
+  billing: BillingHistoryResponse;
+  paymentMethodType = PaymentMethodType;
+  transactionType = TransactionType;
+
+  constructor(
+    protected apiService: ApiService,
+    protected i18nService: I18nService,
+    protected platformUtilsService: PlatformUtilsService,
+    private router: Router
+  ) {}
+
+  async ngOnInit() {
+    if (this.platformUtilsService.isSelfHost()) {
+      this.router.navigate(["/settings/subscription"]);
+    }
```

Comment:
Feels like this belongs better in a guard class of some sort. Creating a tech debt item to figure out this kind of redirects.